skyportal/tests/frontend/test_gcn.py

```diff
@@ -635,3 +635,146 @@ def test_gcn_summary_observations(
 
     finally:
         os.remove(fpath)
+
+
+def test_confirm_reject_source_in_gcn(
+    driver,
+    super_admin_user,
+    super_admin_token,
+    view_only_token,
+    ztf_camera,
+    upload_data_token,
+):
+
+    datafile = f'{os.path.dirname(__file__)}/../../../data/GW190814.xml'
+    with open(datafile, 'rb') as fid:
+        payload = fid.read()
+    data = {'xml': payload}
+
+    status, data = api('POST', 'gcn_event', data=data, token=super_admin_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # wait for event to load
+    for n_times in range(26):
+        status, data = api(
+            'GET', "gcn_event/2019-08-14T21:10:39", token=super_admin_token
+        )
+        if data['status'] == 'success':
+            break
+        time.sleep(2)
+    assert n_times < 25
+
+    # wait for the localization to load
+    params = {"include2DMap": True}
+    for n_times_2 in range(26):
+        status, data = api(
+            'GET',
+            'localization/2019-08-14T21:10:39/name/LALInference.v1.fits.gz',
+            token=super_admin_token,
+            params=params,
+        )
+
+        if data['status'] == 'success':
+            data = data["data"]
+            assert data["dateobs"] == "2019-08-14T21:10:39"
+            assert data["localization_name"] == "LALInference.v1.fits.gz"
+            assert np.isclose(np.sum(data["flat_2d"]), 1)
+            break
+        else:
+            time.sleep(2)
+    assert n_times_2 < 25
+
+    obj_id = str(uuid.uuid4())
+    status, data = api(
+        "POST",
+        "sources",
+        data={
+            "id": obj_id,
+            "ra": 24.6258,
+            "dec": -32.9024,
+            "redshift": 3,
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+
+    status, data = api("GET", f"sources/{obj_id}", token=view_only_token)
+    assert status == 200
+
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': obj_id,
+            'mjd': 58709 + 1,
+            'instrument_id': ztf_camera.id,
+            'flux': 12.24,
+            'fluxerr': 0.031,
+            'zp': 25.0,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            "ra": 24.6258,
+            "dec": -32.9024,
+            "ra_unc": 0.01,
+            "dec_unc": 0.01,
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    driver.get(f'/become_user/{super_admin_user.id}')
+    driver.get('/gcn_events/2019-08-14T21:10:39')
+
+    query_list = driver.wait_for_xpath(
+        '//*[@aria-labelledby="root_queryList-label root_queryList"]', 20
+    )
+    time.sleep(5)
```

Comment:
I noticed that event locally the test was flaky without it. What happens here (and for example also with the favorite source test that is a bit flaky on main) i believe, is that the component rerenders, and sometimes the timing it quite bad. The test finds the xpath, and then when you click its not reachable probably because at the exact moment you tried to click the component rerendered or something similar. Adding a short time.sleep(n) seems to also fix it.